packages/core/src/state-management/job-queue.ts

```diff
@@ -74,6 +107,22 @@ export class JobQueue implements IJobQueue {
     await this.dbConnection.end()
   }
 
+  /**
+   * Updates the job data for a particular job
+   * @param jobId id of the job
+   * @param data data to update
+   * @returns promise that results to a boolean that is true of the
+   */
+  async updateJob(jobId: string, data: object): Promise<void> {
```

Comment:
What happens if we try to update a job that is already running please?

Reply:
From my tests it does not affect an already running job as the data is read at the beginning of the job